src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -76,7 +76,9 @@ public void Dispose()
                     var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
                     var newDocument = e.NewSolution.GetDocument(e.DocumentId);
 
-                    await DocumentChangedAsync(oldDocument, newDocument).ConfigureAwait(false);
+                    // make sure we do this in background thread. we don't care about ordering of events
+                    // we just need to refresh OB at some point if it ever needs to be updated
+                    Task.Run(() => DocumentChangedAsync(oldDocument, newDocument));
```

Comment:
So i'm looking through the code, and i'm not sure why some of it is necessary.  For example, we have the following:

```c#
        private async Task DocumentChangedAsync(Document oldDocument, Document newDocument)
        {
            var oldTextVersion = await oldDocument.GetTextVersionAsync(CancellationToken.None).ConfigureAwait(false);
            var newTextVersion = await newDocument.GetTextVersionAsync(CancellationToken.None).ConfigureAwait(false);

            if (oldTextVersion != newTextVersion)
            {
                UpdateClassAndMemberVersions();
            }
        }
```

But why do we care if the text version changed?  We already just heard that the document changed... so isn't that extra step basically unnecessary?  Why not just immediately call UpdateClassAndMemberVersions from this method?